eventsourcing/src/main/java/org/axonframework/eventsourcing/eventstore/EventCriteriaBuilderEventTypeStage.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventsourcing.eventstore;
+
+import jakarta.annotation.Nonnull;
+
+/**
+ * Interface declaring the possible builder actions for the event type stage of the {@link EventCriteriaBuilder}. After
+ * calling any of the methods defined in this interface, the builder will be in the
+ * {@link EventCriteriaBuilderTagStage}.
+ *
+ * @author Mitchell Herrijgers
+ * @see EventCriteria
+ * @see EventCriteriaBuilder
+ * @since 5.0.0
+ */
+public sealed interface EventCriteriaBuilderEventTypeStage permits EventCriteriaBuilder {
```

Comment:
Since you asked during our meeting, I am going to give my only suggestion for name shortening: `EventTypeCriteriaBuilderStage`

I can't really think of anything else that carries the load sufficiently :shrug:

Reply:
Hmm while the name is nice, it would put the class file in another file order from the EventCriteria class, while now they are grouped together (all start with `EventCriteriaBuilder`). I think I prefer that, so I will leave it for now.